build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/EmbeddedProviderExtension.java

```diff
@@ -36,35 +35,25 @@ void impl(String implName, Project implProject) {
         String projectName = implProject.getName();
         String capitalName = capitalize(projectName);
 
-        String configName = "embedded" + capitalName;
-        Configuration implConfig = project.getConfigurations().create(configName);
+        Configuration implConfig = project.getConfigurations().detachedConfiguration(project.getDependencies().create(implProject));
         implConfig.attributes(attrs -> {
             attrs.attribute(ARTIFACT_TYPE_ATTRIBUTE, DIRECTORY_TYPE);
-            attrs.attribute(IMPL_ATTR, true);
+            attrs.attribute(EmbeddedProviderPlugin.IMPL_ATTR, true);
         });
 
-        var deps = project.getDependencies();
-        deps.registerTransform(UnzipTransform.class,
-            transformSpec -> {
-                transformSpec.getFrom().attribute(ARTIFACT_TYPE_ATTRIBUTE, JAR_TYPE).attribute(IMPL_ATTR, true);
-                transformSpec.getTo().attribute(ARTIFACT_TYPE_ATTRIBUTE, DIRECTORY_TYPE).attribute(IMPL_ATTR, true);
-                transformSpec.parameters(parameters -> parameters.getIncludeArtifactName().set(true));
-            });
-        deps.add(configName, implProject);
-
         String manifestTaskName = "generate" + capitalName + "ProviderManifest";
-        Directory generatedResourcesDir = project.getLayout().getBuildDirectory().dir("generated-resources").get();
+        Provider<Directory> generatedResourcesDir = project.getLayout().getBuildDirectory().dir("generated-resources");
         var generateProviderManifest = project.getTasks().register(manifestTaskName, GenerateProviderManifest.class);
         generateProviderManifest.configure(t -> {
-            t.getManifestFile().set(generatedResourcesDir.file("LISTING.TXT"));
+            t.getManifestFile().set(generatedResourcesDir.get().file("LISTING.TXT"));
             t.getProviderImplClasspath().from(implConfig);
         });
 
         String implTaskName = "generate" + capitalName + "ProviderImpl";
-        Directory implsDir = generatedResourcesDir.dir("impls");
+        Provider<Directory> implsDir = generatedResourcesDir.map(d -> d.dir("impls"));
         var generateProviderImpl = project.getTasks().register(implTaskName, Sync.class);
         generateProviderImpl.configure(t -> {
-            t.setDestinationDir(implsDir.dir(projectName).getAsFile());
+            t.setDestinationDir(implsDir.get().dir(projectName).getAsFile());
```

Comment:
I think we can avoid this call to get as well. Instead of `setDestinationDir` we can call `into()` and then just move everything below in child copy spec of that call.

```java
t.into(generatedResourcesDir, spec -> {
    spec.into("IMPL-JARS/" + implName, childSpec -> {
        childSpec.from(implConfig);
        childSpec.from(generateProviderManifest);
    }
});
```